src/app/services/dimDestinyTrackerService.factory.js

```diff
@@ -25,39 +32,97 @@ function DestinyTrackerService($q,
   const _reviewSubmitter = new ReviewSubmitter($q, $http, dimPlatformService, _trackerErrorHandler, loadingTracker, _reviewDataCache);
   const _reviewReporter = new ReviewReporter($q, $http, dimPlatformService, _trackerErrorHandler, loadingTracker, _reviewDataCache, _userFilter);
 
+  const _d2reviewDataCache = new D2ReviewDataCache();
+  const _d2trackerErrorHandler = new D2TrackerErrorHandler($q, $i18next);
+  const _d2bulkFetcher = new D2BulkFetcher($q, $http, _d2trackerErrorHandler, loadingTracker, _d2reviewDataCache);
+  const _d2reviewsFetcher = new D2ReviewsFetcher($q, $http, _d2trackerErrorHandler, loadingTracker, _d2reviewDataCache, _userFilter, dimPlatformService);
+  const _d2reviewSubmitter = new D2ReviewSubmitter($q, $http, dimPlatformService, _d2trackerErrorHandler, loadingTracker, _d2reviewDataCache);
+  const _d2reviewReporter = new D2ReviewReporter($q, $http, dimPlatformService, _d2trackerErrorHandler, loadingTracker, _d2reviewDataCache, _userFilter);
+
   return {
+    isDestinyOne: function() {
+      return (dimSettingsService.destinyVersion === 1);
+    },
+
+    isDestinyTwo: function() {
+      return (dimSettingsService.destinyVersion === 2);
+    },
+
     reattachScoresFromCache: function(stores) {
-      _bulkFetcher.attachRankings(null,
-                                  stores);
+      if (this.isDestinyOne()) {
+        _bulkFetcher.attachRankings(null,
+                                    stores);
+      }
+      else if (this.isDestinyTwo()) {
+        _d2bulkFetcher.attachRankings(null,
+                                      stores);
+      }
     },
+
     updateCachedUserRankings: function(item,
```

Comment:
At this point it might be easier to expose two services with the same API (as we have in the other code) rather than one service that switches at every point.

Reply:
The problem is that this branching logic would just get pushed out to the edges of the call graph (except for bulk fetch where it's obvious which way we're going).

I started my work with a split of the tracker service, but I wasn't thrilled with what was happening with item component, so I centralized it again.